nima_core/llm_adapter.py

```
@@ -1,113 +1,438 @@
+    ) -> tuple[Optional[str], int, float]:
+        """Execute function with retry logic."""
+        start_time = time.time()
+        last_error = None
+
+        for attempt in range(self.retry_config.max_attempts):
+            try:
+                result = func()
+                if result:
+                    latency = (time.time() - start_time) * 1000
+                    return result, attempt + 1, latency
+            except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)